src/libopensc/pkcs15-pubkey.c

```diff
@@ -1187,19 +1166,15 @@ sc_pkcs15_erase_pubkey(struct sc_pkcs15_pubkey *key)
 			free(key->u.gostr3410.xy.data);
 		break;
 	case SC_ALGORITHM_EC:
+	case SC_ALGORITHM_EDDSA:
+	case SC_ALGORITHM_XEDDSA:
 		if (key->u.ec.params.der.value)
 			free(key->u.ec.params.der.value);
 		if (key->u.ec.params.named_curve)
 			free(key->u.ec.params.named_curve);
 		if (key->u.ec.ecpointQ.value)
 			free(key->u.ec.ecpointQ.value);
```

Comment:
needless if(x) free(x). Please remove the needless `if` here, similarly as it was done in the eddsa case.

Reply:
Just following the coding style in the routine which does the same for RSA, GOST and EC.
``` 
a7a524be1c (Doug Engert         2023-12-08 20:18:51 -0600 1169)         case SC_ALGORITHM_EDDSA:
a7a524be1c (Doug Engert         2023-12-08 20:18:51 -0600 1170)         case SC_ALGORITHM_XEDDSA:
```
I can remove all the needless  `if` if you want..

Reply:
Yes please. This was common antipatern in the past and in a lot of places, probably making people feel safer, but there is not a single case when this would be useful for anything

Reply:
Removed needless if statements before free.
See: add4826da635c783785b5ed33b0f710d9c919873